### PR TITLE
[RFE] SAT-11254 Test Automation

### PR DESCRIPTION
**Pull request by damoore044**
test_positive_block_delete_key_in_use (ui and cli/test_contentcredentials.py)

-UI and CLI test automation
-BZ 2052904 : Blocker now in place when attempting to delete a content credential in use by a product or repository.

Both tests create a new gpg key, a new product with the key, a new repository associated with the product, and attempts to delete the key in use. The tests expect the gpg key to remain associated with the product and repo, and that the gpg, prod, and repo are unmodified.

### CODE REVIEW

**Comment by omkarkhatavkar** [comment]
Can one of the admins verify this patch?

**Review by Griffin-Sullivan — COMMENTED** [review]
After reviewing most of this PR and reading the BZ I feel like you could just test this with one API test. Or maybe 1 API test and 1 CLI test. The fix in the BZ had more to do with the server denying the user from deleting a key that is associated with a product and repo. There's not really anything specific to the CLI and UI about that. What do other's think? I definitely don't want you to feel like you wasted time writing these tests, so if we want to go ahead with these two I can get back to reviewing them.

**Comment by ogajduse** [comment]
@damoore044 Why did you select `6.13.z` as the target branch?

If this PR is not exclusive to 6.13 - the feature you are testing will not be included in Sat 6.14, change the target branch to `master`.
Also if the feature you are focused on exists in the earlier Satellite versions, please assign this PR respective labels to cherry-pick it to `.z` branches.

**Comment by damoore044** [comment]
@Griffin-Sullivan Myself and Cole were originally thinking 1 UI and 1 CLI here. The devs also recommended hammer automation for this one. Do you think the UI testing is redundant in this case? Not opposed to reconsidering that one altogether.

@ogajduse I should have targeted `master` for this pr, I'll update the target branch and labels, thank you

**Comment by Griffin-Sullivan** [comment]
> @Griffin-Sullivan Myself and Cole were originally thinking 1 UI and 1 CLI here. The devs also recommended hammer automation for this one. Do you think the UI testing is redundant in this case? Not opposed to reconsidering that one altogether.

Ok great. I think keeping both is fine I just wanted to hear your thoughts before I review. I'll come back and take another look when this is out of draft.

File: tests/foreman/cli/test_contentcredentials.py [hunk 1]
```diff
@@ -102,6 +102,46 @@ def test_positive_get_info_by_name(module_org):
...
+@pytest.mark.tier1
+def test_positive_block_delete_key_in_use(module_org):
+    """Create a product and single associated repository. Create a new gpg key and associate
+        it with the product and repository. Attempt to delete the gpg key in use
+
+    :id: 022555fd-e6f2-4c95-80d7-cae26993ca8f
+
+    :expectedresults: deletion of the gpg key is blocked by raising CLIReturnCodeError 70,
+        the key is not deleted, and remains associated with the product and repository
+
```
Comment (ColeHiggins2) [inline tests/foreman/cli/test_contentcredentials.py:114]: maybe add a BZ number to this test too ?

File: tests/foreman/cli/test_contentcredentials.py [hunk 2]
```diff
@@ -102,6 +102,46 @@ def test_positive_get_info_by_name(module_org):
...
+    product = make_product({'organization-id': module_org.id})
+    repo = make_repository({'product-id': product['id']})
+    gpg_key = make_content_credential(
+        {'key': VALID_GPG_KEY_FILE_PATH, 'name': name, 'organization-id': module_org.id}
+    )
+    # Associate repo with the product, gpg key with product and repo
+    Product.update(
+        {'gpg-key-id': gpg_key['id'], 'id': product['id'], 'organization-id': module_org.id}
+    )
+
```
Comment (ColeHiggins2) [inline tests/foreman/cli/test_contentcredentials.py:126]: Lets get rid of random spacing

File: tests/foreman/cli/test_contentcredentials.py [hunk 3]
```diff
@@ -102,6 +102,46 @@ def test_positive_get_info_by_name(module_org):
...
+    # Attempt to delete gpg in use
+    with pytest.raises(CLIReturnCodeError) as e:
+        ContentCredential.delete({'name': name, 'organization-id': module_org.id})
+    assert 'Cannot delete' in str(e.value)
+
```
Comment (ColeHiggins2) [inline tests/foreman/cli/test_contentcredentials.py:131]: same here, and so on

**Pull request closed without merging**